Libraries/Image/RCTImageLoader.m

```diff
@@ -480,6 +470,18 @@ - (RCTImageLoaderCancellationBlock)loadImageWithURLRequest:(NSURLRequest *)image
   __block void(^cancelLoad)(void) = nil;
   __weak RCTImageLoader *weakSelf = self;
 
+  // Load handlers are executed before the generic caching logic so they are responsible
+  // for caching the images if necessary.
+  id<RCTImageURLLoader> loadHandler = [self imageURLLoaderForURL:imageURLRequest.URL];
```

Comment:
Yes we have to skip the URLCache queue and also avoid caching the resulting image as this is already handled by UIImage imageNamed, we definetly don't want to cache then twice. Afaik memory should not be an issue since it only applies to local images and they are usually small icons or a few large images.